catalyst_voices/packages/internal/catalyst_voices_blocs/test/session/session_cubit_test.dart

```diff
@@ -20,6 +23,9 @@ void main() {
   late SessionCubit sessionCubit;
 
   setUpAll(() {
+    final store = InMemorySharedPreferencesAsync.empty();
+    SharedPreferencesAsyncPlatform.instance = store;
+
```

Comment:
It's a bit not clear which dependency needs it. Shared prefs are not used in SessionCubit directly so unless you know what all dependencies of session cubit are doing then you can only guess where this ends up.

Would it be possible to mock dependencies of SessionCubit in a way that this is not needed?

For example if we write a test for the service and the service depends on repository then we mock the repository so that it doesn't try communicating with a real API. Here we would need to mock the dependency that tries talking to shared prefs so that it doesn't attempt to. Or you can pass an instance of mocked shared prefs directly via constructor to this dependency so it's clear which one needs shared prefs

Reply:
I've made secureStorage and sharedPreferences required dependency so it's more explicit why we need it. Like we disused